platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java

```diff
@@ -37,4 +37,9 @@
      * Returns command type (i.e. 'maven')
      */
     String getType();
+
+    /**
+     * Returns command preview URL. (i.e. 'http://awesome.com:8080/context')
+     */
+    String getPreviewUrl();
```

Comment:
Preview url as a string is useless. [Here](https://codenvy.slack.com/archives/general/p1451318006000395) Tyler proposed another scheme. If we are OK to implement this, we need also realize how to set file: preview object.
Preview object as  string won't allow us to implement proposed strategies.